### PR TITLE
store/mockstore: Add DeleteRange to mock MvccStore and MVCCLevelDB

### PR DESCRIPTION
**Pull request by MyonKeminta**
Support DeleteRange in mock store.
Not implemented yet in MVCCLevelDB.

I'm trying to add DeleteRange interface to ticlient. DeleteRange function of mock store is required to do unit tests.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@coocood PTAL

File: store/mockstore/mocktikv/mvcc.go [hunk 1]
```diff
@@ -743,6 +744,27 @@ func (s *MvccStore) BatchResolveLock(startKey, endKey []byte, txnInfos map[uint6
 	return nil
 }
 
+// DeleteRange deletes all keys in the given range.
+func (s *MvccStore) DeleteRange(startKey, endKey []byte) error {
```
Comment (tiancaiamao) [inline store/mockstore/mocktikv/mvcc.go]: MvccStore is obsolete, you should implement MVCCLevelDB.

Comment (MyonKeminta) [reply store/mockstore/mocktikv/mvcc.go]: Already done. PTAL

**Comment by MyonKeminta** [comment]
@tiancaiamao @coocood PTAL

**Comment by tiancaiamao** [comment]
LGTM @coocood

**Comment by MyonKeminta** [comment]
@coocood PTAL

**Comment by zimulala** [comment]
@MyonKeminta 
Do we need to update `SupportDeleteRange`?

**Comment by MyonKeminta** [comment]
In `ddl/delete_range.go`:
```
// start implements delRangeManager interface.
func (dr *delRange) start() {
	if !dr.storeSupport {
		dr.d.wait.Add(1)
		go dr.startEmulator()
	}
}
```
It seems that it didn't implement how to do if delete range is supported. Changing `SupportDeleteRange` in `tikv/kv.go` makes lots of unit tests broken. There may be still much work to do before changing that field.

**Pull request merged**